tests/KubernetesClient.Tests/ThreadSafeStoreTests.cs

```diff
@@ -0,0 +1,140 @@
+using k8s.cache;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace k8s.Tests
+{
+    public class ThreadSafeStoreTests
+    {
+        private static readonly TimeSpan TestTimeout = TimeSpan.FromSeconds(150);
+
+        private readonly ITestOutputHelper testOutput;
+
+        public ThreadSafeStoreTests(ITestOutputHelper testOutput)
+        {
+            this.testOutput = testOutput;
+        }
+
+        [Fact]
+        public void Simple()
+        {
+            var store = new ThreadSafeStore();
+            var kObject = new KubernetesObject();
+
+            var key = "default/kind4";
+            var apiVersion = "v4";
+            var kind = "kind4";
+            kObject.ApiVersion = apiVersion;
+            kObject.Kind = kind;
+
+            // Simple add test
+            store.Add(key, kObject);
+            var val = store.Get(key);
+            Assert.Equal(val.ApiVersion, apiVersion);
+            Assert.Equal(val.Kind, kind);
+
+            // Add again which should raise and exception
+            var raisedExcection = false;
+            try
+            {
+                store.Add(key, kObject);
+            }
+            catch (Exception)
+            {
+                raisedExcection = true;
+            }
+            Assert.True(raisedExcection);
+            val = store.Get(key);
+            Assert.Equal(val.ApiVersion, apiVersion);
+            Assert.Equal(val.Kind, kind);
+
+            // Update existing key
+            var newApiVersion = "v4update";
+            var newKind = "kind4update";
+            var newKObject = new KubernetesObject();
+            newKObject.ApiVersion = newApiVersion;
+            newKObject.Kind = newKind;
+
+            var oldVal = store.Update(key, newKObject);
+            // Check if the object matches the new key.
+            val = store.Get(key);
+            Assert.Equal(val.ApiVersion, newApiVersion);
+            Assert.Equal(val.Kind, newKind);
+            // Check the old value
+            Assert.Equal(oldVal.ApiVersion, apiVersion);
+            Assert.Equal(oldVal.Kind, kind);
+
+            key = "non existant key";
+            store.Update(key, kObject);
+            val = store.Get(key);
+            // Check if the updated value (by add) is correct.
+            Assert.Equal(val.ApiVersion, apiVersion);
+            Assert.Equal(val.Kind, kind);
+
+            // Delete
+            store.Delete(key);
+            val = store.Get(key);
+            Assert.Null(val);
+
+            raisedExcection = false;
+            // Delete again 
+            try
+            {
+                store.Delete(key);
+            }
+            catch (Exception)
+            {
+                raisedExcection = true;
+            }
+
+            val = store.Get(key);
+            Assert.Null(val);
+        }
+
+        [Fact]
+        public void MultiThread()
+        {
+            var store = new ThreadSafeStore();
+            Action<object> action = (object o) =>
```

Comment:
```suggestion
            var action = (int o) =>
```

the caller calls using int as  param